### DISCUSSION TITLE
Is it possible to make everything in a model a pointer and let nullable fields use guregu/null package?

### DISCUSSION THREAD

**Discussion by logic-and-math** [go-jet/jet#360] [Q&A] [answered]
Hi, first I want to say: incredible package!
I would love to set up the generator so my models have every field as a pointer and nullable fields are also pointers but with corresponding null value from the https://github.com/guregu/null package.
I see the model field type change in the generator customisation example, but I'm wondering if changing every model field to a pointer will break something in serialization/deserialisation?

**Comment by go-jet** [comment, marked as answer]
Hi @logic-and-math ,
> I'm wondering if changing every model field to a pointer will break something in serialization/deserialisation?

No, it should be fine. It should work regardless of the presence of the pointer.

**Reply by logic-and-math** [reply]
Ty @go-jet !
Yeah I've been testing it and it seems fine. Currently facing an issue with null values, since I would like to differentiate when a value is not selected in the query, when a value is null, and when a value is not null. If I make a *null.String jet will read null values as nil instead of null.String with valid = false (but that could be unrelated to jet, not sure). But do you know if that's possible to override ?

**Reply by go-jet** [reply]
> If I make a *null.String jet will read null values as nil instead of null.String with valid = false (but that could be unrelated to jet, not sure). But do you know if that's possible to override ?

No, that can't be overridden. Why do you need double null fields?

**Reply by logic-and-math** [reply]
@go-jet 
Like 80% of my endpoints return a model from the database (in the same form) with a few joins and different field selections.
The way I see it is that if I can add omitempty to every model during generation, I can define most of my endpoints using jet models(without defining which fields are for which response). And since the fields which are not selected get a zerovalue they will not be included in the response json. So in a way it could save me a lot of response definitions, and adding/removing fields is simply changing the query.
If golang had omitzero instead of omitempty, it would work(by adding a present flag to the nullable type). Im assuming using a custom serializer would work also, but have to figure out what's the easiest way to implement it. Either a generator after yours to add the method to each struct or something like that.

**Reply by houtn11** [reply]
@logic-and-math Check this discussion - https://github.com/go-jet/jet/discussions/429, might be related.
